.github/workflows/ci-e2e.yaml

```diff
@@ -82,3 +84,61 @@ jobs:
           name: playwright-report
           path: playwright-report/
           retention-days: 5
+
+  accessibility_test:
+    name: Run accessibility tests
+    #if: github.event.schedule == '0 21 1 * *'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+
+      - name: Install yarn
+        run: |
+          corepack enable
+          yarn set version 3.2.1
+          yarn config set nodeLinker 'node-modules'
+          yarn config set npmScopes.kausal.npmRegistryServer "${{ secrets.NPM_REGISTRY_SERVER }}"
+          yarn config set npmScopes.kausal.npmAlwaysAuth true
+          yarn config set npmScopes.kausal.npmAuthIdent ${{ secrets.NPM_AUTH_IDENT }}
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Playwright install
+        run: node_modules/.bin/playwright install --with-deps
+
+      - name: Set TEST_PLAN_IDENTIFIERS for each job
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            echo "TEST_PLAN_IDENTIFIERS=${{ github.event.inputs.plan_identifiers }}" >> $GITHUB_ENV
+            echo "FRONTEND_BASE_URL=${{ github.event.inputs.frontend_base_url }}" >> $GITHUB_ENV
+          else
+            echo "TEST_PLAN_IDENTIFIERS=${{ secrets.TEST_PLAN_IDENTIFIERS }}" >> $GITHUB_ENV
+            IFS=',' read -ra PLAN_IDS <<< "$TEST_PLAN_IDENTIFIERS"
+            for PLAN_ID in "${PLAN_IDS[@]}"
+            do
+              FRONTEND_BASE_URL="http://${PLAN_ID}.watch-test.kausal.tech"
+              echo "Running tests for plan $PLAN_ID with FRONTEND_BASE_URL=$FRONTEND_BASE_URL"
+              echo "APLANS_API_BASE_URL=https://api.watch.kausal.tech/v1" >> $GITHUB_ENV
+            done
+          fi
+
+      - name: Running Playwright accessibility tests
+        run: node_modules/.bin/playwright test accessibility.spec.ts
+        env:
+          TEST_PLAN_IDENTIFIERS: ${{ env.TEST_PLAN_IDENTIFIERS }}
+          TEST_PAGE_BASE_URL: ${{ env.FRONTEND_BASE_URL }}
+          APLANS_API_BASE_URL: ${{ env.APLANS_API_BASE_URL }}
+
+      - uses: actions/upload-artifact@v3
+        if: always()
```

Comment:
It looks like a lot of copy-pasting happened here, not sure if there is a way to refactor these parts and reuse?